### PR TITLE
Fix `generate-docs` Command Sections Parsing

### PR DESCRIPTION
**Pull request by kgal-pan**
## Related Issues
fixes:https://jira-dc.paloaltonetworks.com/browse/CIAC-10635, https://jira-dc.paloaltonetworks.com/browse/XSUP-36639

## Description
The PR includes the following changes to the `generate-docs` command:
- A fix/improved implementation of finding the commands' sections within the integration `README.md` which is used when updating modified commands.
- A new flag, `--force`, that forces the generation of the new document (instead of updating the existing one based on source control). This allows the user to have more flexibility with usage.
- Add [blocknotes](https://github.com/orgs/community/discussions/16925) to command `README`:

image

- Fix an issue were a modified command name was treated as a new command and appended to the `README`.

### CODE REVIEW

File: demisto_sdk/commands/generate_docs/README.md [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-Generate a README file for an Integration, Script or a Playbook.
+Generate a `README.md` file for an Integration, Script or a Playbook.
+
+## Use Cases
 
-**Use-Cases**
 This command is used to create a documentation file for Cortex XSOAR content files.
 
-**Arguments**:
+## Arguments
```
Comment (samuelFain) [inline demisto_sdk/commands/generate_docs/README.md:9]: Are all the `**Example**` to `##Example` changes fine from a BC point of view?
These are the changes that enables support for code like: `{HEADER_TYPE.H3}`, correct?

**Review by samuelFain — CHANGES_REQUESTED** [review]
Nice!
@kgal-pan  Please review my very few comments :)
Also, it would be much appreciated if you could document these changes in [here](https://confluence-dc.paloaltonetworks.com/display/DemistoContent/demisto-sdk+command+related+information).

File: .changelog/4303.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Improved implementation of finding the commands' sections within the integration README.md
```
[the remark this answers is not in the document]
Comment (kgal-pan) [reply .changelog/4303.yml]: https://github.com/demisto/demisto-sdk/pull/4303/commits/87670b3948d41842523d581f2032329df5a7aa99

**Review by samuelFain — APPROVED** [review]
Nice!

**Pull request merged**